scheduler/publisher.py

```diff
@@ -36,15 +36,17 @@ def check_setup(connection, channel, RMQHOSTNAME, RMQPORT):
         )
 
 # Function to format message before sending to Notifier / Project Microservice
-def format_message(resource_id, type, milestone_id):
+def format_message(resource_id, type, milestone_id, payment_id):
+
     new_data = {
         "project_id": resource_id,
-        "milestone_id": milestone_id
+        "milestone_id": milestone_id,
+        "payment_id": payment_id
     }
     return json.dumps({"resource_id": resource_id, "type": type, "data": new_data})
 
 # Function that republishes tasks that failed
-def publishTask(event, project_id, milestone_id):
+def publishTask(event, project_id, milestone_id, payment_intent_id=None):
     type = ""
```

Comment:
Why change the type here? Why not just pass the event as the type?

Reply:
I have updated the code to simplify the logic. do check and let me know if there should be anymore changes